plugins/TapTempo/TapTempoView.cpp

```diff
@@ -0,0 +1,169 @@
+	font.setPointSize(24);
+
+	m_tapButton = new QPushButton();
+	m_tapButton->setFixedSize(200, 200);
+	m_tapButton->setFont(font);
+	m_tapButton->setText(tr("0"));
+
+	auto precisionCheckBox = new QCheckBox(tr("Precision"));
+	precisionCheckBox->setFocusPolicy(Qt::NoFocus);
+	precisionCheckBox->setToolTip(tr("Display in high precision"));
+	precisionCheckBox->setText(tr("Precision"));
+
+	auto muteCheckBox = new QCheckBox(tr("0.0 ms"));
+	muteCheckBox->setFocusPolicy(Qt::NoFocus);
+	muteCheckBox->setToolTip(tr("Mute metronome"));
+	muteCheckBox->setText(tr("Mute"));
+
+	m_msLabel = new QLabel();
+	m_msLabel->setFocusPolicy(Qt::NoFocus);
+	m_msLabel->setToolTip(tr("BPM in milliseconds"));
+	m_msLabel->setText(tr("0 ms"));
+
+	m_hzLabel = new QLabel();
+	m_hzLabel->setFocusPolicy(Qt::NoFocus);
+	m_hzLabel->setToolTip(tr("Frequency of BPM"));
+	m_hzLabel->setText(tr("0.0000 hz"));
+
+	auto resetButton = new QPushButton(tr("Reset"));
+	resetButton->setFocusPolicy(Qt::NoFocus);
+	resetButton->setToolTip(tr("Reset counter and sidebar information"));
+
+	auto syncButton = new QPushButton(tr("Sync"));
+	syncButton->setFocusPolicy(Qt::NoFocus);
+	syncButton->setToolTip(tr("Sync with project tempo"));
+
+	auto optionLayout = new QVBoxLayout();
+	optionLayout->addWidget(precisionCheckBox);
+	optionLayout->addWidget(muteCheckBox);
+
+	auto bpmInfoLayout = new QVBoxLayout();
+	bpmInfoLayout->addWidget(m_msLabel, 0, Qt::AlignHCenter);
+	bpmInfoLayout->addWidget(m_hzLabel, 0, Qt::AlignHCenter);
+
+	auto sidebarLayout = new QHBoxLayout();
+	sidebarLayout->addLayout(optionLayout);
+	sidebarLayout->addLayout(bpmInfoLayout);
+
+	auto buttonsLayout = new QHBoxLayout();
+	buttonsLayout->addWidget(resetButton, 0, Qt::AlignCenter);
+	buttonsLayout->addWidget(syncButton, 0, Qt::AlignCenter);
+
+	auto mainLayout = new QVBoxLayout(this);
+	mainLayout->addWidget(m_tapButton, 0, Qt::AlignCenter);
+	mainLayout->addLayout(buttonsLayout);
+	mainLayout->addLayout(sidebarLayout);
+
+	connect(m_tapButton, &QPushButton::pressed, this, [this, muteCheckBox]() {
+		m_plugin->onBpmClick();
+		updateLabels();
+
+		if (!muteCheckBox->isChecked())
+		{
+			const auto timeSigNumerator = Engine::getSong()->getTimeSigModel().getNumerator();
+			m_plugin->m_numTaps % timeSigNumerator == 0
+				? Engine::audioEngine()->addPlayHandle(new SamplePlayHandle("misc/metronome02.ogg"))
+				: Engine::audioEngine()->addPlayHandle(new SamplePlayHandle("misc/metronome01.ogg"));
+		}
+	});
+
+	connect(resetButton, &QPushButton::pressed, this, [this]() { closeEvent(nullptr); });
+
+	connect(precisionCheckBox, &QCheckBox::toggled, [this](bool checked) {
+		m_plugin->m_showDecimal = checked;
+		updateLabels();
+	});
+
+	connect(syncButton, &QPushButton::clicked, this, [this]() {
+		const auto& tempoModel = Engine::getSong()->tempoModel();
+		if (m_plugin->m_bpm < tempoModel.minValue() || m_plugin->m_bpm > tempoModel.maxValue()) { return; }
+		Engine::getSong()->setTempo(m_plugin->m_bpm);
```

Comment:
```suggestion
		Engine::getSong()->setTempo(std::round(m_plugin->m_bpm));
```
I think the bpm should be rounded first, so that values like 119.9 aren't truncated to 119.